bug: Docs build fails due to Starlight social config format change

## Summary

The documentation site build fails with an Astro/Starlight configuration error related to the `social` field format.

## Error

```
18:53:17 [ERROR] [@astrojs/starlight] An unhandled error occurred while running the "astro:config:setup" hook
[AstroUserError] Invalid config passed to starlight integration
  Hint:
    social: Expected type `"object"`, received `"array"`
```

## Cause

The `social` config in `docs/astro.config.mjs` uses an array format:

```js
social: [
  {
    icon: "github",
    label: "GitHub",
    href: "https://github.com/0pilatos0/bitbucket-cli",
  },
],
```

But newer versions of Starlight expect an object format:

```js
social: {
  github: "https://github.com/0pilatos0/bitbucket-cli",
},
```

## Steps to Reproduce

```bash
cd docs
bun run build
```

## Suggested Fix

Update `docs/astro.config.mjs` to use the object format for the `social` config, or check Starlight version compatibility.

--- a/.changeset/starlight-social-config.md
+++ b/.changeset/starlight-social-config.md
@@ -1,0 +1,5 @@
+---
+"@pilatos/bitbucket-cli": patch
+---
+
+Fix Starlight social config format in documentation site. Changed from array format to object format to match newer Starlight version requirements.

--- a/docs/astro.config.mjs
+++ b/docs/astro.config.mjs
@@ -20,13 +20,9 @@ export default defineConfig({
           },
         },
       ],
-      social: [
-        {
-          icon: "github",
-          label: "GitHub",
-          href: "https://github.com/0pilatos0/bitbucket-cli",
-        },
-      ],
+      social: {
+        github: "https://github.com/0pilatos0/bitbucket-cli",
+      },
       sidebar: [
         {
           label: "Getting Started",

